pkg/k8s/apis/cilium.io/v2/cew_types.go

```diff
@@ -0,0 +1,99 @@
+//  Copyright 2020 Authors of Cilium
```

Comment:
You'll need to add two more things:

1) Make the agent wait for this new CRD, see commit e91f803c3
2) As Andre mentioned, you'll need to add the RBAC permissions for the agent to access this new resource, see https://github.com/cilium/cilium/blob/17c4f80ea32495ecd62c4f1a39455c45b2210f6c/install/kubernetes/cilium/templates/cilium-agent-clusterrole.yaml#L90

Reply:
Cilium agents are not accessing CiliumExternalWorkload CRDs, so IMO agent RBAC permissions do not need to be amended. Only `clustermesh-apiserver` reads CEW CRDs :-)

Reply:
@christarazi Also, I'm not sure if `clustermesh-apiserver` needs to wait for CRDs or not? It is watching on a number of CRDs, as well as creating and updating CRDs.

Reply:
@jrajahalme If that's the case, cilium agent does not need any RBAC permissions to access the new resource. However, `clustermesh-apiserver` needs those RBAC permissions.

>@christarazi Also, I'm not sure if `clustermesh-apiserver` needs to wait for CRDs or not? It is watching on a number of CRDs, as well as creating and updating CRDs.

It depends, if `clustermesh-apiserver` depends on those CRDs then it needs to be sure they are available, and if not, it needs to wait for them.

Reply:
OK, I'll need to add a wait for the CRDs before starting any of the CRD watchers in `clustermesh-apiserver`.